bionic/module_reloader.py

```diff
@@ -0,0 +1,24 @@
+import builtins
+from importlib import reload
+from sys import modules as module_registry
+
+
+def recursive_reload(module):
+    original_import = builtins.__import__
+    already_reloaded = set()
+
+    def custom_import(name, globals=None, locals=None, fromlist=[], level=0):
+        if name in module_registry:
```

Comment:
Before checking `name` in `module_registry`, would you need to change it using `globals` and `locals` when the `level` is relative instead of being absolute?

Reply:
Similarly, when we have submodules in `fromlist`, should we just reload those submodules instead of reloading the entire module?

Reply:
My guess is that `__import__()` will be called on those submodules anyways, and `custom_import` will be called. Hence, there is probably no need to call `reload` on those submodules?

Reply:
But if someone imports `a.b` using `from a import b`, would we also reload other submodules of `a` this way (that are not needed / used)?

Reply:
I am not entirely sure, but I think the `importlib.reload` works on a per-module basis and hence it may update all variables that are associated with the reloaded module? This is not worse than before since we were also calling `reload`.